pyre/check.py

```
@@ -1774,6 +1775,200 @@ def default_binary(backend):
+    subtrees, and a gate reading mtimes then blocks a build that is in fact
+    current. Hashing the roughly one thousand inputs (about 1GB, most of it
+    the LLBC) costs under a second, against the multi-minute build
+    `--no-build` exists to skip.
+
+    Computed after a build rather than before it: `cargo` may rewrite
+    `Cargo.lock`, which is itself an input, so the state that produced the
+    artefact is the state that exists once the build has finished. Nothing
+    else a build writes is in the set — that is what keeps the two orders
+    equivalent for every other input.
+    """
+    global _BUILD_INPUTS_FINGERPRINT
```

Ruff (0.16.1)

[warning] Using the global statement to update `_BUILD_INPUTS_FINGERPRINT` is discouraged

(PLW0603)